math/krishnamurthy.go

```diff
@@ -0,0 +1,30 @@
+// filename : krishnamurthy.go
+// description: A program which contains the function that returns true if a given number is Krishnamurthy number or not.
+// details: A number is a Krishnamurthy number if the sum of all the factorials of the digits is equal to the number.
+// Ex: 1! = 1, 145 = 1! + 4! + 5!
+// author(s): [GooMonk](https://github.com/GooMonk)
+// see krishnamurthy_test.go
+package math
+
+// IsKrishnamurthyNumber returns if the provided number n is a Krishnamurthy number or not.
+func IsKrishnamurthyNumber[T int32 | int64 | int | uint32 | uint64 | uint](n T) bool {
```

Comment:
I don't think there is any real benefit to using this instead of the `contrainst.Integer` constraint. Is there a reason you changed it?

Reply:
Hi @tjgurwara99, sorry for the late reply. I thought of using int8 and int16 would give wrong results for the number 40585, so thought only allowing the function call for int32 and int64 would be better.

Reply:
Then how about `type myInt int64`? Do you think your algorithm would work for this?

The answer is no 😄 - the point of using a generic constraint to use the same algorithm on similar types but if the algorithm by nature is not benefiting from a generic type then don't use generics.

Also, I don't see why it won't work on `int8` and `int64`. Can you elaborate on why you think it won't work on them? (I even tested them and they seem to be working - so wanted to confirm with you)

Reply:
I was thinking about the overflow error it would give for the larger numbers but it would not make sense for the user to store large numbers in the low bounded int types. I will use the contrainst.Integer again and also create work on the other suggestions as mentioned by you. Thank you for the insight.